temporal-opentracing/src/main/java/io/temporal/opentracing/internal/OpenTracingContextAccessor.java

```diff
@@ -0,0 +1,70 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.opentracing.internal;
+
+import com.google.gson.reflect.TypeToken;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.propagation.*;
+import io.temporal.api.common.v1.Payload;
+import io.temporal.common.converter.DataConverter;
+import io.temporal.common.interceptors.Header;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class OpenTracingContextAccessor {
+  private static final String TRACER_HEADER_KEY = "_tracer-data";
+  private static final Type HASH_MAP_STRING_STRING_TYPE =
+      TypeToken.getParameterized(HashMap.class, String.class, String.class).getType();
+
+  public static void writeSpanContextToHeader(
+      SpanContext spanContext, Header header, Tracer tracer) {
+    Map<String, String> serializedSpanContext = serializeSpanContextToMap(spanContext, tracer);
+    Optional<Payload> payload = DataConverter.getDefaultInstance().toPayload(serializedSpanContext);
```

Comment:
Shouldn't we use[ tracer.inject ](https://opentracing.io/guides/java/inject-extract/) and `tracer.extract` instead of DataConverter?

Reply:
Not really. tracer.inject and tracer.extract are working with TEXT_MAPs. So, we convert a span to a TEXT_MAP using tracer.inject inside serializeSpanContextToMap, but after that we need to serialize this map for temporal headers and we use DataConverter for that. tracer.inject is used here, it just happened inside serializeSpanContextToMap.

Reply:
What about using BINARY format to avoid JSON type serialization? You still would need to use DataConverter to create the Payload, but it will use the binary payload as it is.

Reply:
I don’t mind, I don’t see problems with both approaches. If you prefer Binary - let’s go with Binary. Will delete guava and rework back to Hash Maps then :) Temporal Go SDK uses TEXT_MAP + converter approach though. You sure you want it differently between SDKs?

Reply:
I don't have a strong opinion on this. If you feel that map is more debuggable let's keep it as it is.

Reply:
Yeah, me too. Let's go with Map. It looks like it's the most used and referenced implementation and it's also in agreement with how things are done in Go SDK.